File: Java/src/main/java/domain/SeatId.java
package domain;

public class SeatId {
    public int seat_number;
    private final String coachId;


    public SeatId(int seat_number, String coachId) {
        this.seat_number = seat_number;
        this.coachId = coachId;
    }

    public String getSeatId() {
        return coachId + seat_number;
    }
}

File: csharp/src/Domain/SeatId.cs
namespace TrainKata.Domain
{
    public struct SeatId
    {
        public CoachId CoachId { get; }
        public SeatNumber SeatNumber { get; }

        public SeatId(CoachId coachId, SeatNumber seatNumber)
        {
            CoachId = coachId;
            SeatNumber = seatNumber;
        }

        public override string ToString()
        {
            return $"{SeatNumber}{CoachId}";
        }
    }
}